DevEdu.API/DevEdu.API/Controllers/NotificationController.cs

```diff
@@ -1,61 +1,62 @@
 ﻿using System.Collections.Generic;
+using System.ComponentModel;
 using AutoMapper;
 using DevEdu.API.Models.InputModels;
+using DevEdu.API.Models.OutputModels;
+using DevEdu.Business.Services;
 using DevEdu.DAL.Models;
-using DevEdu.DAL.Repositories;
+using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Mvc;
 
 namespace DevEdu.API.Controllers
 {
     [ApiController]
     [Route("api/[controller]")]
-    public class NotificationController
+    public class NotificationController : Controller
     {
-        private readonly INotificationRepository _notificationRepository;
+        private readonly INotificationService _notificationService;
         private readonly IMapper _mapper;
-
-        public NotificationController(IMapper mapper, INotificationRepository notificationRepository)
+        public NotificationController(IMapper mapper, INotificationService notificationService)
         {
-            _notificationRepository = notificationRepository;
+            _notificationService = notificationService;
             _mapper = mapper;
         }
 
         //  api/notification/5
         [HttpGet("{id}")]
         public NotificationDto GetNotification(int id)
         {
-            return _notificationRepository.GetNotification(id);
+            return _notificationService.GetNotification(id);
         }
 
         //  api/notification/by-user/1
         [HttpGet("by-user/{userId}")]
         public List<NotificationDto> GetAllNotificationsByUserId(int userId)
         {
-            return _notificationRepository.GetNotificationsByUser(userId);
+            return _notificationService.GetNotificationsByUserId(userId);
```

Comment:
I think we need additional endpoints:
GetAllNotificationsByRoleId
GetAllNotificationsByGroupId